### PR TITLE
Implemented file multi selection.

### LINKED ISSUE #18: Multiple file selection?
I would like to use this library in my app, but this feature is missing. As of now I rely on rfd, but it would be nice if egui_file supported this.

### PR DESCRIPTION
**Pull request by GuillaumeSchmid**
This is a simple implementation of multiple file selection.
closes #18 

image

The shift key can be used to select ranges of files and ctrl can add/remove from selection without reseting it.
I am currently using  it in my image processing application to load batches of images at once.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -51,6 +55,9 @@ pub struct FileDialog {
   /// Files in directory.
   files: Result<Vec<FileInfo>, Error>,
 
+  /// Flag selected files in multi-select mode.
+  multi_selection: Vec<bool>,
```
Comment (Barugon) [inline src/lib.rs:59]: Instead of a separate vector, why not add a `selected` member to `FileInfo`?

File: src/lib.rs [hunk 2]
```diff
@@ -308,6 +341,11 @@ impl FileDialog {
         self.confirm();
       }
     }
+    else if self.multi_select_enabled {
+      if self.dialog_type == DialogType::OpenFile {
+        self.confirm();
```
Comment (Barugon) [inline src/lib.rs:346]: These two terms can be combined.

**Comment by Barugon** [comment]
Other than the separate vector, it looks good. Thanks for implementing this.

**Comment by GuillaumeSchmid** [comment]
Thank you for your feedback, I implemented your recommandations.

**Review by Barugon — APPROVED** [review]

**Pull request merged**